### ISSUE TITLE
JSON collapse

### ISSUE THREAD

**Issue by mastacheata** [odin-public/RESTClient-APS#42]
I'd like to replace the current JSON display library "KelpJSONView" with [jJsonViewer](https://github.com/Shridhad/jjsonviewer), which among other features can collapse and expand JSON object and array children.
This would significantly enhance the readability of long responses and is probably an alternative to your pagination idea from #31 

I'll attempt this for my personal build now and will submit a PR once I'm confident it works.

**Comment by gear54rus** [comment]
Mentioning #43

**Comment by gear54rus** [comment]
@mastacheata I can see that your directive here might not be properly followed by new JSON engine: https://i.imgur.com/hlhIsUc.png

https://i.imgur.com/Rl0lmdz.png

No line numbers are displayed. Can you please look what's up with that? 

I think I'll merge this then... even though it seems resource intensive when collapsing.

**Comment by gear54rus** [comment]
Also, if we press 'SEND' many times while staying on 'Highlighted' screen, this is what we see: https://i.imgur.com/DWh6ZKx.png

**Comment by mastacheata** [comment]
That is a known bug.
I thought I had updated the pr to fix that already, so that should only be a matter of finding the right commit. 
Will look into it on Monday, I guess. 
-- 
Gruß, 
Benedikt Bauer 

Am 8. Juli 2016 11:33:24 MESZ, schrieb Paul <notifications@github.com>:

**Comment by mastacheata** [comment]
Ok the second issue is fixed. Missed adding that fix in the rebased PR.

The first one is a bit tricky, though. Either I can find a way to display continuous line numbers through multiple stacked lists or I'll need to change the jjsonviewer to print the line numbers directly.

**Comment by mastacheata** [comment]
So, the newest commit should have the line numbers and a minor performance boost.
I don't think, there's room for much to gain anymore.

New build can be found on my own build server:
https://teamcity.ymoq.de/guestAuth/app/rest/builds/id:376/artifacts/content/restclient-aps.xpi

**Comment by gear54rus** [comment]
**This will be merged as it is.** Still I've noticed something minor:

![tyx7tjl](https://cloud.githubusercontent.com/assets/8073599/16792600/a6a589ea-48eb-11e6-9ad4-3a03ed0899b7.gif)

The first line has broken indentation. And it seems to be due to it being handled similarly to all subsequent lines by the engine, which leaves some space on the left for collaps/expand sign. That seems internal for that engine, but if you know how to fix that, you still have time before I merge this.

**Comment by gear54rus** [comment]
Also what about these commits? can you please make sure that the 'jjsonviewer' branch includes everything we need?

![myainy3](https://cloud.githubusercontent.com/assets/8073599/16799622/92d989a2-4913-11e6-8c40-db88a7a151dc.png)

**Comment by mastacheata** [comment]
The lower 3 were squashed before creating the Jjson branch. The last ones are part of the merge request (you can see that github adds any additional commits into the same branch to the pr after creating it. 
-- 
Gruß, 
Benedikt Bauer 

Am 13. Juli 2016 12:05:40 MESZ, schrieb Paul <notifications@github.com>:

**Comment by mastacheata** [comment]
I'm working on the develop branch and only cherry pick commits into the Jjson branch when development is done. That is because my develop branch contains unrelated changes as well. 
-- 
Gruß, 
Benedikt Bauer 

Am 13. Juli 2016 12:05:40 MESZ, schrieb Paul <notifications@github.com>:

**Comment by gear54rus** [comment]
So I should merge only what's in PR, correct? What about the first line, any info?

**Comment by mastacheata** [comment]
That is correct. I'll check that first line issue any minute now.

**Comment by mastacheata** [comment]
The issue about the top-level elements collapser is now fixed in a way.
The collapsing/expanding logic doesn't work on the top-level element. (It also makes no sense to collapse that as you wouldn't see anything anymore)
So I removed the marker and the different cursor from the first expander/collapser.
Change is in the jjsonviewer branch and XPI build available at https://teamcity.ymoq.de/guestAuth/app/rest/builds/id:384/artifacts/content/restclient-aps.xpi
